File: gestion-tournoi-frontend/src/app/stade.ts
export interface StadeResponse {
  id: number;
  nom: string;
  ville: string;
  heureOuverture: string; // format "HH:mm:ss"
  heureFermeture: string;
  matchIds: number[];
}

export interface CreateStade {
  nom: string;
  ville: string;
  heureOuverture: string;
  heureFermeture: string;
}

export interface UpdateStade {
  id?: number;
  nom?: string;
  ville?: string;
  heureOuverture?: string;
  heureFermeture?: string;
}

File: gestionDesTournois Backend/src/main/java/org/example/football/entites/Stade.java
package org.example.football.entites;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.Locale;
import java.util.TimerTask;

@Entity
@Table(name = "stade")
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString(exclude = "matchs")
@EqualsAndHashCode(exclude = "matchs")
public class Stade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;
    private String ville;

    private LocalTime heureOuverture;
    private LocalTime heureFermeture;

    @OneToMany(mappedBy = "stade", cascade = CascadeType.ALL)
    private List<Match> matchs;
}
